src/main/java/betterquesting/client/gui2/GuiQuestLines.java

```diff
@@ -518,4 +526,77 @@ private void refreshClaimAll()
+        UUID playerUUID = QuestingAPI.getQuestingUUID(mc.thePlayer);
+        CanvasQuestSearch canvasQuestSearch = new CanvasQuestSearch(new GuiTransform(GuiAlign.FULL_BOX, new GuiPadding(0, 32, 8, 24), 0)) {
+            @Override
+            protected boolean addResult(QuestSearchEntry entry, int index, int cachedWidth) {
+                PanelButtonCustom buttonContainer = new PanelButtonCustom(new GuiRectangle(0, index * 32, cachedWidth, 32, 0), 2);
+                buttonContainer.setCallback(panelButtonCustom -> {
+                    if (!buttonContainer.isActive()) return;
+                    thisGui.initPanel();
+                    openQuestLine(entry.getQuestLineEntry());
+                    int selectedQuestId = entry.getQuest().getID();
+                    Optional<PanelButtonQuest> targetQuestButton = cvQuest.getQuestButtons().stream().filter(panelButtonQuest -> panelButtonQuest.getStoredValue().getID() == selectedQuestId).findFirst();
+                    targetQuestButton.ifPresent(panelButtonQuest ->{
+                        GuiTextureColored newTexture = new GuiTextureColored(panelButtonQuest.txFrame,
+                                new GuiColorPulse(
+                                        new GuiColorStatic(255, 220, 115, 255),
+                                        new GuiColorStatic(255, 191, 0, 255),
+                                        1, 0
+                                ));
+                        panelButtonQuest.setTextures(newTexture, newTexture, newTexture);
+                    });
+                });
+                buttonContainer.setActive(QuestCache.isQuestShown(entry.getQuest().getValue(), playerUUID, mc.thePlayer));
+                this.addPanel(buttonContainer);
+
+                PanelButtonQuest questButton = new PanelButtonQuest(new GuiRectangle(2, 2, 28, 28), 0, "", entry.getQuest());
+                questButton.setCallback(value -> {
+                    if (!questButton.isActive()) return;
+                    thisGui.initPanel();
+                    openQuestLine(entry.getQuestLineEntry());
+                    GuiHome.bookmark = new GuiQuest(thisGui, entry.getQuest().getID());
+                    mc.displayGuiScreen(GuiHome.bookmark);
+                });
+                buttonContainer.addPanel(questButton);
+
+                buttonContainer.addPanel(new PanelGeneric(new GuiRectangle(36, 2, 14, 14, 0), new OreDictTexture(1F, entry.getQuestLineEntry().getValue().getProperty(NativeProps.ICON), false, true)));
+
+                PanelTextBox chapterName = new PanelTextBox(new GuiRectangle(56, 6, cachedWidth - 56, 16), QuestTranslation.translate(entry.getQuestLineEntry().getValue().getProperty(NativeProps.NAME)));
+                buttonContainer.addPanel(chapterName);
+
+                PanelTextBox questName = new PanelTextBox(new GuiRectangle(36, 20, cachedWidth - 40, 16), QuestTranslation.translate(entry.getQuest().getValue().getProperty(NativeProps.NAME)));
```

Comment:
All of the PanelTextBox -> .addPanel calls can be placed in a multi-purpose seperate method